test/test_testing.py

```diff
@@ -683,5 +685,224 @@ def test_calculate_2_shards_against_optimal_shards(self):
                 self.assertEqual(sorted_tests, sorted_shard_tests)
 
 
+class TestAsserts(TestCase):
```

Comment:
nit why don't we create `TestAsserts` and `TestAssertsMultiDevice` ? seems like
1. all but 2 is decorated with `@onlycuda` and 
2. seems like in the future we don't want to just check against cuda, but other device types too?

Reply:
That is a good idea. Since it is not blocking I'll put it at the end of the stack to avoid fixing all the conflicts.